core/src/main/scala/spinal/core/sim/package.scala

```diff
@@ -356,50 +358,49 @@ package object sim {
     }
   }
 
-  /**
-    * Add implicit function to Bits
-    */
-  implicit class SimBitsPimper(bt: Bits) {
+  protected abstract class RandomizableBitVector(bt: BitVector, longLimit: Int) {
+    protected val width = bt.getWidth
 
-    def randomize(): Unit = {
-      val width = bt.getWidth
-      if(width < 64){
-        bt #= Random.nextLong() & ((1l << width) - 1)
-      }else {
-        bt #= BigInt(width, Random)
+    def randomize(): BigInt = {
+      if (width < longLimit) {
+        val l = randomizedLong()
+        bt #= l
+        l
+      } else {
+        val bi = randomizedBigInt()
+        bt #= bi
+        bi
       }
     }
 
+    def randomizedBigInt(): BigInt
+
+    def randomizedLong(): Long
+
+    def randomizedInt(): Int
+  }
+
+  /**
+   * Add implicit function to Bits
+   */
+  implicit class SimBitsPimper(bt: Bits) extends RandomizableBitVector(bt, 64) {
     def randomizedBigInt() = {
-      val width = bt.getWidth
       BigInt(width, Random)
     }
     def randomizedLong() = {
-      val width = bt.getWidth
       assert(width < 64)
       Random.nextLong() & ((1l << width) - 1)
     }
     def randomizedInt() = {
-      val width = bt.getWidth
       assert(width < 32)
       Random.nextInt() & ((1 << width) - 1)
     }
   }
 
   /**
-    * Add implicit function to UInt
-    */
-  implicit class SimUIntPimper(bt: UInt) {
-
-    def randomize(): Unit = {
-      val width = bt.getWidth
-      if(width < 64){
-        bt #= Random.nextLong() & ((1l << width) - 1)
-      }else {
-        bt #= BigInt(width, Random)
-      }
-    }
-
+   * Add implicit function to UInt
+   */
+  implicit class SimUIntPimper(bt: UInt) extends RandomizableBitVector(bt, 64) {
```

Comment:
`extends SimBitsPimper(bt)` with no body? Or even, as suggested above, defining defaults for functions in the base class, and override functions only for `SInt`?

Reply:
sure, I forgot to commit that before I pushed and had it lying around here